registry/zookeeper/registry.go

```diff
@@ -371,6 +390,23 @@ func (r *zkRegistry) register(c common.URL) error {
 		dubboPath = fmt.Sprintf("/dubbo/%s/%s", r.service(c), (common.RoleType(common.CONSUMER)).String())
 		logger.Debugf("consumer path:%s, url:%s", dubboPath, rawURL)
 
+	case common.ROUTER:
+		//todo
+		dubboPath = fmt.Sprintf("/dubbo/%s/%s", c.Service(), common.DubboNodes[common.ROUTER])
+		r.cltLock.Lock()
+		err = r.client.Create(dubboPath)
```

Comment:
if u wanna lock a func, u'd better use lambda and defer-unlock to lock it safely. What's more, u can use it as follows: r.cltLock.Lock()<br/>clt := r.client<br/>r.cltLock.Unlock()<br/>if clt == nil { }<br/> err = clt.Create(dubboPath)